projects/js-packages/components/components/text/index.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.module.scss';
+import type { H3Props, TextProps, TitleProps } from './types';
+import { BOX_MODEL_VALUES, VARIANTS_MAPPING } from './constants';
+
+/**
+ * Text component.
+ *
+ * @param {TextProps} props - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+const Text: React.FC< TextProps > = ( {
+	variant = 'body',
+	children,
+	component,
+	className,
+	...componentProps
+} ) => {
+	const Component = ( component || VARIANTS_MAPPING[ variant ] || 'span' ) as React.FC< TextProps >;
+
+	// Build Styles module CSS classnames.
+	const boxModelClasses = useMemo( () => {
+		return BOX_MODEL_VALUES.reduce( ( acc, value ) => {
+			if ( typeof componentProps[ value ] !== 'undefined' ) {
+				acc += styles[ `${ value }-${ componentProps[ value ] }` ] + ' ';
+				// pick spacing prop. Do not pass down to Component.
+				delete componentProps[ value ];
+			}
+			return acc;
+		}, '' );
+	}, [ componentProps ] );
+
+	return (
+		<Component
+			className={ classNames( styles.reset, styles[ variant ], className, boxModelClasses ) }
+			{ ...componentProps }
+		>
+			{ children }
+		</Component>
+	);
+};
+
+export default Text;
+
+/**
+ * Heading component - Medium size.
+ *
+ * @param {TextProps} props      - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+export const H2: React.FC< TextProps > = ( { children, ...componentProps } ) => (
+	<Text variant="headline-medium" mb={ 3 } { ...componentProps }>
+		{ children }
+	</Text>
+);
+
+/**
+ * Heading component - Small size,
+ *
+ * @param {H3Props} props        - Component props.
+ * @returns {React.ReactElement} - JSX.Element
+ */
+export const H3: React.FC< H3Props > = ( { children, weight = 'bold', ...componentProps } ) => {
+	const _weight = ( weight === 'bold' ? '' : weight ) as string;
```

Comment:
I suggest avoiding `_` as a const starting name, this is not being a used pattern. But we could discuss more

Reply:
Thanks, what name do you suggest?

Reply:
`weightStr`?

Reply:
`normalizedWeight` | `currentWeight` | `h3Weight`

Reply:
Done, thank you.